agent_nodes.tf

```
@@ -46,7 +46,7 @@ locals {
         [
           "--node-ip ${agent.ip}",
           "--node-name '${try(agent.name, key)}'",
-          "--server https://${local.root_server_ip}:6443",
+          "--server https://${local.root_advertise_ip_curl}:6443",
```

Updated variable name

```suggestion
          "--server https://${local.root_advertise_ip_k3s}:6443",
```